### DISCUSSION TITLE
dynamic lift works but not correctly - Elegoo Saturn 4K

### DISCUSSION THREAD

**Discussion by Paimon667** [sn4k3/UVtools#618] [Q&A] [answered]
I did a test with dynamic elevation with a certain file I created in Tinkercad, two rectangles, one with a very large area and the other with a very small area, one above the other, so there is a sudden transition between one rectangle and another, in the transition between a "big" layer and a small layer it doesn't behave correctly, when making the last big layer it doesn't use the correct elevation and speed parameters, it uses the one of the next layer, which are parameters of a small layer, being so it does less elevation than it should on a large layer, I have an Elegoo Saturn 4k, the file was prepared in Chitubox, but I also tested it with Tango Slicer

**Comment by sn4k3** [comment, marked as answer]
Provide that file so I can check

**Reply by Paimon667** [reply]
here ->
inside there is the .ctb file that I used in the printer and also the .stl object

[Sizzling Curcan-Crift (10).zip](https://github.com/sn4k3/UVtools/files/10197629/Sizzling.Curcan-Crift.10.zip)

**Reply by sn4k3** [reply]
Based on UVtools information that file should:

- Lift 7mm on layer 0 and 1 @ 70mm/min
- Lift 10mm on layer 2 to 5 @ 180mm/min
- Lift 2.3mm on layer 6 (First smaller squre) to 11 @ 180mm/min

So when changing from big square to small square, that first small square layer should lift 2.3mm before print.
Is that correct with what you observed?

**Reply by Paimon667** [reply]
No, when making the last layer of the big rectangle it lifts 2.3mm and not 10mm as it should, I think it makes the current layer with the parameters of the next layer.

When looking at the Elegoo Saturn screen when finishing "drawing" layer 5, which is the last large rectangle, it already increments the current layer to 6, I think that's why it uses the settings for layer 6 and not 5 for rise

**Reply by sn4k3** [reply]
What you just described is the same as I.
Layer 6 is lifting 2.3mm before print the first small rectangle instead of the 10mm, right?

![image](https://user-images.githubusercontent.com/113281/206787960-38218bf2-cfda-4b8e-b046-f49add61f53a.png)

The thing is: Current layer will lift the value it show before print, and yes it is wrong because it come from a bigger mass, it may be a bug on code, I will check

**Reply by Paimon667** [reply]
Here is a video I made showing in practice how this code is behaving on my printer, it should be more useful than me trying to explain

https://www.youtube.com/shorts/cFh3lTH2nOg

obs ::
I noticed from the video, I think that my printer treats layer 0 as if it didn't exist, it really starts from layer 1, so it goes from layer 1 to 12 and not from 0 to 11 as in the UVTools program, I could be wrong , but from what I understand it must be something related to that

**Reply by sn4k3** [reply]
Yeah as told it's the same both described.
It's a problem in the logic, should fix in the next release.

> I noticed from the video, I think that my printer treats layer 0 as if it didn't exist, it really starts from layer 1, so it goes from layer 1 to 12 and not from 0 to 11 as in the UVTools program

UVtools uses indexes, 0 started
Chitubox and printer may use 1 started which is why you see that difference

**Reply by sn4k3** [reply]
Fixed on last version, please check.

**Reply by Paimon667** [reply]
I did the test with the same file and now it works perfectly, finally I can use this amazing function of this tool, thank you very much
